include/swift/SIL/SILInstruction.h

```diff
@@ -523,12 +523,20 @@ class SILInstruction
 private:
   /// Predicate used to filter OperandValueRange.
   struct OperandToValue;
+  /// Predicate used to filter TransformedOperandValueRange.
+  struct OperandToTransformedValue;
 
 public:
   using OperandValueRange =
       OptionalTransformRange<ArrayRef<Operand>, OperandToValue>;
+  using TransformedOperandValueRange =
+      OptionalTransformRange<ArrayRef<Operand>, OperandToTransformedValue>;
+
   OperandValueRange
   getOperandValues(bool skipTypeDependentOperands = false) const;
+  TransformedOperandValueRange
+  getOperandValues(std::function<SILValue(SILValue)> transformFn,
+                   bool skipTypeDependentOperands) const;
```

Comment:
Ran into an overload resolution issue with clang when I use a default value for `skipTypeDependentOperands`.
See https://godbolt.org/z/17514r. Leaving it as is for now.

Reply:
I think the issue is that a pointer picks the implicit bool conversion over the converting constructor in `std::function`. I think it's fine the way it is, but one way to fix it would be to explicitly construct a `std::function` in the caller.

Reply:
Yea, it can be a mild surprise. I didn't want to add it to avoid any confusion for new callers.